ophyd/utils/epics_pvs.py

```diff
@@ -325,3 +326,13 @@ def records_from_db(fn):
         ret.append((rtype, record))
 
     return ret
+
+def raise_if_disconnected(fcn):
+    '''Decorator to catch attempted access to disconnected EPICS channels.'''
+    @functools.wraps(fcn)
+    def wrapper(self, *args, **kwargs):
+        if self.connected:
+            return fcn(self, *args, **kwargs)
+        else:
+            raise Exception('{} is not connected'.format(self.name))
```

Comment:
Patterns like this can be very helpful, we use it heavily in bluesky https://github.com/NSLS-II/bluesky/blob/master/bluesky/run_engine.py#L618 to deal with cleaning up with a variety of different failure modes.